mint/run/core/healthcheck/main.go

```diff
@@ -199,9 +199,10 @@ func testPrometheusEndpoint(endpoint string) {
 func main() {
 	endpoint := os.Getenv("SERVER_ENDPOINT")
 	secure := os.Getenv("ENABLE_HTTPS")
-	endpoint = "http://" + endpoint
 	if secure == "1" {
 		endpoint = "https://" + endpoint
+	} else {
+		endpoint = "http://" + endpoint
 	}
 
```

Comment:
Not sure what does this fix? @mclarkson - AFAICS there is nothing fixed here.

Reply:
Without the fix and using http:
> http://minio_endpoint

Without the fix using https it calls:
> https://http://minio_endpoint

Because enpoint already has 'http://' in it when it creates the new https endpoint.

Reply:
Ah I see to make it cleaner I think it should be implemented a bit more cleanly

Reply:
Do you mean you think this fix is to make the code cleaner?
Without this fix the health check *does not work* when targetting HTTPS.

It does not work because it tries to contact: "https://http://endpoint", which obviously is not a valid address.

Reply:
We can take the fix and work on rewriting it later.